test/qa/test_grpc.py

```diff
@@ -0,0 +1,79 @@
+import threading
+import sh
+import grpc
+from collections.abc import Sequence
+from lib import daemon, info, logging, login
+from lib.protobuf.daemon import (common_pb2, service_pb2_grpc, state_pb2)
+
+NORDVPND_SOCKET = 'unix:///run/nordvpn/nordvpnd.sock'
+
+
+def setup_function():  # noqa: ARG001
+    daemon.start()
+    login.login_as("default")
+    logging.log()
+
+
+def teardown_function():  # noqa: ARG001
+    logging.log(data=info.collect())
+    logging.log()
+
+    sh.nordvpn.logout("--persist-token")
+    sh.nordvpn.set.defaults()
+    daemon.stop()
+
+
+def test_multiple_state_subscribers():
+    expected_states = [
+        state_pb2.ConnectionState.CONNECTING,
+        state_pb2.ConnectionState.CONNECTED,
+    ]
+
+    num_threads = 5
+    threads = []
+    results = {}
+
+    threads = [threading.Thread(target=lambda i=i: results.update(
+        {i: collect_state_changes(len(expected_states), ['connection_status'])})) for i in range(num_threads)]
+
+    [thread.start() for thread in threads]
+    sh.nordvpn.connect()
+    [thread.join() for thread in threads]
+
+    for i in range(num_threads):
+        assert all(a.connection_status.state == b for a, b in zip(
+            results[i], expected_states, strict=True))
+
+
+# TODO
```

Comment:
Leftover? Or is there something to do here?

Reply:
Leftover. Thanks for noticing!